sdk/src/trace/recordable.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+
+/**
+ * Maintains a representation of a span in a format that can be processed by a recorder.
+ *
+ * This class is thread-compatible.
+ */
+class Recordable
```

Comment:
Do you think that we need to represent parenting on this level? Given this interface, I don't see how an exporter can determine a parent span or recordable.

Reply:
Not all the methods are filled in (because we don't have all the types and methods in the api yet).

But there will be some kind of SpanContext type with a corresponding, SetSpanContext or something similar as well as AddLink methods. Those would pass the parenting information.